dao/src/test/java/com/fitgoal/dao/impl/utils/DocumentUtils.java

```diff
@@ -0,0 +1,58 @@
+package com.fitgoal.dao.impl.utils;
+
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+import org.bson.Document;
+import org.testcontainers.shaded.org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.fitgoal.dao.util.AuditDtoFields.DATE;
+
+@UtilityClass
+class DocumentUtils {
+
+    private static final SimpleDateFormat SIMPLE_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+    List<Document> parseDocuments(@NonNull String resourceName) {
+        try {
+            String docJson = IOUtils.resourceToString(resourceName, StandardCharsets.UTF_8);
+
+            Object content = Document.parse("{\"json\":" + docJson + "}").get("json");
+            List<Document> documents = (List<Document>) content;
+            return documents
```

Comment:
This whole flow looks a bit weird, all you want to do is to add additional field to the existing documents, there is no need to create a new list.
So you can have instead just

`            documents.forEach(fixDate());
            return documents; `
`
    @NotNull
    private static Consumer<Document> fixDate() {
        return document -> document.put(DATE, convertDate(document.getString(DATE)));
    }
`

Reply:
Refactored